### PR TITLE
Enable async backing for system parachains

### PR DESCRIPTION
**Pull request by s0me0ne-unkn0wn**
This enables async backing with 6 sec block times for all the system parachains on both Kusama and Polkadot.

### CODE REVIEW

**Comment by s0me0ne-unkn0wn** [comment]
@bkontur bridges fail tests because of constants defined [here](https://github.com/paritytech/polkadot-sdk/blob/994003854363b3c5fd0d7343f93aa1e54edb1ad0/bridges/chains/chain-bridge-hub-cumulus/src/lib.rs#L45). How should we handle that? Do we need first to release a new version of `bh-bridge-hub-cumulus` with new values and then bump dependencies here?

**Comment by bkontur** [comment]
> @bkontur bridges fail tests because of constants defined [here](https://github.com/paritytech/polkadot-sdk/blob/994003854363b3c5fd0d7343f93aa1e54edb1ad0/bridges/chains/chain-bridge-hub-cumulus/src/lib.rs#L45). How should we handle that? Do we need first to release a new version of `bh-bridge-hub-cumulus` with new values and then bump dependencies here?

I think there is no need to change `bh-bridge-hub-cumulus`, I added there stuff for async backing here: https://github.com/paritytech/polkadot-sdk/pull/2966.

Please, try to change this `BlockWeights` -> `BlockWeightsForAsyncBacking` [here](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/bridge-hubs/bridge-hub-kusama/primitives/src/lib.rs#L55), [here](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/bridge-hubs/bridge-hub-polkadot/primitives/src/lib.rs#L52), [here](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/bridge-hubs/bridge-hub-kusama/src/bridge_to_polkadot_config.rs#L402) and [here](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/bridge-hubs/bridge-hub-polkadot/src/bridge_to_kusama_config.rs#L398)

**Comment by s0me0ne-unkn0wn** [comment]
@bkontur hmmm, it's not in 0.8.0 `bh-bridge-hub-cumulus` crate version which is a workspace dependency... Can I just bump it to the latest 0.10.0? Or better wait until you bump all the crate versions for the next release?

**Comment by bkontur** [comment]
> @bkontur hmmm, it's not in 0.8.0 `bh-bridge-hub-cumulus` crate version which is a workspace dependency... Can I just bump it to the latest 0.10.0? Or better wait until you bump all the crate versions for the next release?

hmm, sorry, I didn't check the version. I would not bump here (you could run into the other issues with incompatibility...),
I would just copy [BlockWeightsForAsyncBacking](https://github.com/paritytech/polkadot-sdk/pull/2966/files#diff-d09a78041a7b450915d1aa6016d4200d5ceb5b56ec463e1bb37797552789947cR102-R120)) and [MAXIMUM_BLOCK_WEIGHT_FOR_ASYNC_BACKING](https://github.com/paritytech/polkadot-sdk/pull/2966/files#diff-d09a78041a7b450915d1aa6016d4200d5ceb5b56ec463e1bb37797552789947cR60) directly [here](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/bridge-hubs/bridge-hub-kusama/primitives/src/lib.rs) and [here](https://github.com/polkadot-fellows/runtimes/blob/main/system-parachains/bridge-hubs/bridge-hub-kusama/primitives/src/lib.rs) with some `TODO: consider removing when 1.10.0 bump` added here: https://github.com/polkadot-fellows/runtimes/issues/186

File: system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs [hunk 1]
```diff
@@ -196,7 +196,7 @@ impl pallet_timestamp::Config for Runtime {
 	/// A timestamp: milliseconds since the unix epoch.
 	type Moment = u64;
 	type OnTimestampSet = Aura;
-	type MinimumPeriod = ConstU64<{ SLOT_DURATION / 2 }>;
+	type MinimumPeriod = ConstU64<0>;
```
Comment (ermalkaleci) [inline system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs:199]: is this a tmp value or is it going to be zero?

Comment (s0me0ne-unkn0wn) [reply system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs:199]: IIUC, we don't need to set the minimum period anymore. With async backing at higher velocities, we may produce more blocks per slot, and nothing stops us from producing more than two blocks per slot, so this limit doesn't make sense, and I set it to zero here. But maybe some of the senior fellows will correct me.

Comment (ermalkaleci) [reply system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs:199]: If that's the case then I need to figure out how to deal with it on chopsticks

File: system-parachains/bridge-hubs/bridge-hub-kusama/primitives/src/lib.rs [hunk 2]
```diff
@@ -20,16 +20,48 @@
...
 	dispatch::DispatchClass,
+	parameter_types,
 	sp_runtime::{MultiAddress, MultiSigner},
+	weights::constants,
 };
+use frame_system::limits;
 use sp_runtime::{FixedPointNumber, FixedU128, RuntimeDebug, Saturating};
 
+// TODO: Remove this as soon as the `bp-bridge-hub-cumulus` version is bumped to 0.10.0
+// See https://github.com/polkadot-fellows/runtimes/issues/186
```
[the remark this answers is not in the document]
Comment (s0me0ne-unkn0wn) [reply system-parachains/bridge-hubs/bridge-hub-kusama/primitives/src/lib.rs:38]: > No we don't want to import this from the polkadot-sdk repo

But we always did. What has changed? I mean, I'd like to add a sensible comment in the code on why we decided to change it.

[the remark this answers is not in the document]
Comment (bkchr) [reply system-parachains/bridge-hubs/bridge-hub-kusama/primitives/src/lib.rs:38]: Always? 

We have ported the primitives to this repo, where they should live. If there are still imports of variables from the polkadot-sdk repo, they should be moved to this repo.

[the remark this answers is not in the document]
Comment (s0me0ne-unkn0wn) [reply system-parachains/bridge-hubs/bridge-hub-kusama/primitives/src/lib.rs]: Alright, removed the comments and tasks from #186.

**Comment by bkchr** [comment]
Before we are doing this, we need to review that all these chains are capable of running with 6seconds. I mean that stuff that maybe has timeout measured in blocks would be broken by this.

**Comment by xlc** [comment]
We can't change the block time without ensuring every pallet used by each runtime are dynamic block time friendly
I think we need to have a checklist somewhere
Related https://github.com/paritytech/polkadot-sdk/issues/3635 https://github.com/paritytech/polkadot-sdk/issues/3268

**Comment by s0me0ne-unkn0wn** [comment]
Okay, so it's not that easy. Maybe I could create separate PRs for different parachains, then? Do we have any parachains that definitely can be switched to 6s block times without side effects?

The point is, we have teams in the ecosystem waiting for at least something running with 6s block times on mainnets. The sooner we show it works, the sooner they'll catch up. And sorting out the timing issue may take quite some time, I'm afraid.

**Comment by xlc** [comment]
there is good chance no existing ecosystem parachain can enable 6s block time because none of the existing governance pallet supports changing block time

**Comment by s0me0ne-unkn0wn** [comment]
> there is good chance no existing ecosystem parachain can enable 6s block time because none of the existing governance pallet supports changing block time

But governance does support runtime upgrades, and we have `SLOT_DURATION` as a runtime constant. That's not a concern IIUC. As well as all the stuff using proper time constants (those `HOURS`, `MINUTES` etc.) derived from the block time. Only stuff explicitly using block numbers to measure time is affected.

**Comment by bkchr** [comment]
> Only stuff explicitly using block numbers to measure time is affected.

What Bryan meant is that governance logic uses the block number to measure time. Most of the stuff is using/used block number for time. This is the problem we are facing here.

**Comment by s0me0ne-unkn0wn** [comment]
> Most of the stuff is using/used block number for time. This is the problem we are facing here.

Oh I see. Still, it's a shame we noticed the problem, we could otherwise get two fellowship salaries a month :rofl:

**Comment by s0me0ne-unkn0wn** [comment]
Okay, I went through the existing system parachains, and indeed, most of them require resolving timing issues. But what about the `people` chain? Are there any obstacles to running it with 6 sec block times? Maybe I'm missing something, but I don't really see anything that would break.

**Pull request closed without merging**